src/downloadAzureProject/setupProjectFolder.ts

```diff
@@ -0,0 +1,103 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { WebSiteManagementModels } from '@azure/arm-appservice';
+import { RequestPrepareOptions } from '@azure/ms-rest-js';
+import * as extract from 'extract-zip';
+import * as querystring from 'querystring';
+import * as vscode from 'vscode';
+import { IActionContext, parseError } from 'vscode-azureextensionui';
+import { initProjectForVSCode } from '../commands/initProjectForVSCode/initProjectForVSCode';
+import { ProjectLanguage } from '../constants';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { SlotTreeItemBase } from "../tree/SlotTreeItemBase";
+import { getNameFromId } from '../utils/azure';
+import { requestUtils } from '../utils/requestUtils';
+import { getRequiredQueryParameter, HandleUriActions } from './handleUri';
+
+export async function setupProjectFolder(uri: vscode.Uri, vsCodeFilePathUri: vscode.Uri, context: IActionContext): Promise<void> {
+    const parsedQuery: querystring.ParsedUrlQuery = querystring.parse(uri.query);
+    const resourceId: string = getRequiredQueryParameter(parsedQuery, 'resourceId');
+    const devContainerName: string = getRequiredQueryParameter(parsedQuery, 'devcontainer');
+    const language: string = getRequiredQueryParameter(parsedQuery, 'language');
+    const action: string = getRequiredQueryParameter(parsedQuery, 'action');
+    const downloadAppContent: boolean = action === HandleUriActions.downloadContentAndSetupProject;
+
+    const toBeDeletedFolderPathUri: vscode.Uri = vscode.Uri.joinPath(vsCodeFilePathUri, 'temp');
+
+    try {
+        const functionAppName: string = getNameFromId(resourceId);
+        const downloadFilePath: string = vscode.Uri.joinPath(toBeDeletedFolderPathUri, `${functionAppName}.zip`).fsPath;
+
+        await vscode.window.withProgress({ location: vscode.ProgressLocation.Notification, title: localize('settingUpFunctionAppLocalProjInfoMessage', `Setting up project for function app '${functionAppName}' with language '${language}'.`) }, async () => {
+            if (downloadAppContent) {
+                // NOTE: We don't want to download app content for compiled languages.
```

Comment:
Yes, I will update to the C#Script ProjectLanguage. And with that we would be downloading content as well in our flow.
Yup, for C# (aka not script) we don't need to worry about in VS Code for our scenario

Reply:
So doesn't that mean we only need to support "downloadContentAndSetupProject", not "setupProject"?

Reply:
We need to support both as we need setupProject for Java as Java is a compiled language

Reply:
For our extension, C# is used 8 times more than Java. I would prefer we don't support _any_ compiled language because I don't think it's useful in the context of this feature, but if we're going to do a compiled language it just feels like we should do C# before Java.

Fwiw, our C#Script usage is basically non-existent (C# is used 100 times more than C#Script), so that's debate-ably not worth supporting, either.

Reply:
This is all behind a feature flag. We can pivot on ideas as we get feedback on usage from customers internally in the team.